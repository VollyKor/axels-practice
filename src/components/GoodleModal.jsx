import { useState, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { GoogleMap, useLoadScript, Marker } from '@react-google-maps/api';
import Modal from 'react-bootstrap/Modal';
import Button from 'react-bootstrap/Button';

import { setGeo } from '../redux/ducks/geo';

const containerStyle = {
    width: '100%',
    height: '100%',

    minWidth: '200px',
    minHeight: '300px',
};

const options = {
    disableDefaultUI: true,
    zoomControl: true,
};

const libraries = ['places'];

const GoogleModal = ({ fillForm, ...props }) => {
    const [marker, setMarker] = useState(() => ({
        lat: 0,
        lng: 0,
    }));

    const geo = useSelector((state) => state.geo);
    const dispatch = useDispatch();

    useEffect(() => {
        setMarker({ ...geo });
    }, [dispatch, geo]);

    const { isLoaded, loadError } = useLoadScript({
        googleMapsApiKey: process.env.REACT_APP_GOOGLE_API_KEY,
        libraries,
        language: 'en',
    });

    if (loadError) return 'Google Map Error';
    if (!isLoaded) return 'Google Map Loading';

    return (
        <Modal
            {...props}
            size="lg"
            aria-labelledby="contained-modal-title-vcenter"
            centered
        >
            <Modal.Header closeButton>
                <Modal.Title id="contained-modal-title-vcenter">
                    Choose Your Address
                </Modal.Title>
            </Modal.Header>
            <Modal.Body>
                <GoogleMap
                    mapContainerStyle={containerStyle}
                    center={geo}
                    zoom={14}
                    options={options}
                    onClick={(event) => {
                        setMarker({
                            lat: event.latLng.lat(),
                            lng: event.latLng.lng(),
                        });
                    }}
                >
                    <Marker position={geo && marker} />
                </GoogleMap>
            </Modal.Body>
            <Modal.Footer>
                <Button
                    onClick={() => {
                        dispatch(setGeo(marker));
                        props.onHide();
                        fillForm(marker);
                    }}
                >
                    Submit
                </Button>
                <Button
                    variant="secondary"
                    onClick={() => {
                        props.onHide();
                    }}
                >
                    Close
                </Button>
            </Modal.Footer>
        </Modal>
    );
};

export default GoogleModal;
