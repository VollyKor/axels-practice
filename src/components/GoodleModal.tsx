import { useState, useEffect } from 'react';

import { useDispatch, useSelector } from 'react-redux';
import { GoogleMap, useLoadScript, Marker } from '@react-google-maps/api';
import Modal from 'react-bootstrap/Modal';
import Button from 'react-bootstrap/Button';

import { setGeo } from '../redux/ducks/geo';
import { Libraries } from '@react-google-maps/api/dist/utils/make-load-script-url';

const GoogleMapsStyles = {
    width: '100%',
    height: '100%',

    minWidth: '200px',
    minHeight: '300px',
};

const options = {
    disableDefaultUI: true,
    zoomControl: true,
};

const libraries: Libraries = ['places'];

interface Props {
    fillForm: TfillForm;
    onHide: Function;
    show: boolean;
}

const GoogleModal = ({ fillForm, ...props }: Props) => {
    const googleKey = process.env.REACT_APP_GOOGLE_API_KEY || 'unvalidKey';

    const [marker, setMarker] = useState(() => ({
        lat: 0,
        lng: 0,
    }));

    const geo = useSelector(({ geo }: RootState) => geo);
    const dispatch = useDispatch();

    useEffect(() => {
        setMarker({ ...geo });
    }, [dispatch, geo]);

    const { loadError } = useLoadScript({
        googleMapsApiKey: googleKey,
        libraries,
        language: 'en',
    });

    if (loadError) return <p>Google Map Error</p>;

    return (
        <Modal
            {...props}
            size="lg"
            aria-labelledby="contained-modal-title-vcenter"
            centered
        >
            <Modal.Header closeButton>
                <Modal.Title id="contained-modal-title-vcenter">
                    Choose Your Address
                </Modal.Title>
            </Modal.Header>

            <Modal.Body>
                <GoogleMap
                    mapContainerStyle={GoogleMapsStyles}
                    center={geo}
                    zoom={14}
                    options={options}
                    onClick={(event) => {
                        if (event.latLng && event.latLng) {
                            setMarker({
                                lat: event.latLng.lat(),
                                lng: event.latLng.lng(),
                            });
                        }
                    }}
                >
                    <Marker position={geo && marker} />
                </GoogleMap>
            </Modal.Body>

            <Modal.Footer>
                <Button
                    onClick={() => {
                        dispatch(setGeo(marker));
                        props.onHide();
                        fillForm(marker);
                    }}
                >
                    Submit
                </Button>
                <Button
                    variant="secondary"
                    onClick={() => {
                        props.onHide();
                    }}
                >
                    Close
                </Button>
            </Modal.Footer>
        </Modal>
    );
};

export default GoogleModal;
