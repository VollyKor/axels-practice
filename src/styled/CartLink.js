import styled from 'styled-components';
import { IoCartOutline } from 'react-icons/io5';

import { Wrapper } from './utils';

export const Cart = () => (
    <Wrapper>
        <span className="mr-1">cart</span>
        <IoCartOutline size="25px" />
        <p>1</p>
    </Wrapper>
);

export const CartWrapper = styled.div`
    position: relative;
    color: var(--cart-link);
`;
export const CartText = styled.span``;

export const CartNotification = styled.div`
    position: absolute;
    top: 0;
    right: -2px;

    width: 12px;
    height: 12px;

    color: var(--white-color);
    background-color: var(--pink-color);
    border-radius: 50%50%;

    font-size: 9px;
`;
