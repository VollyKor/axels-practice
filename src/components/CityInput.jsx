import { useState } from 'react';
import Form from 'react-bootstrap/Form';
import { BiTargetLock } from 'react-icons/bi';

import { FindButton } from '../styled/CityInput';
import GoogleModal from './GoodleModal';

export default function CityInput({
    inputRef,
    fillForm,
    value = '',
    ...props
}) {
    const [show, setShow] = useState(false);

    return (
        <div className="position-relative">
            <Form.Group controlId="city" className="position-relative">
                <Form.Control
                    type="text"
                    placeholder="City"
                    ref={inputRef}
                    value={value}
                    {...props}
                />
            </Form.Group>

            <FindButton type="button" onClick={() => setShow(!show)}>
                <BiTargetLock size="25px" color="#5a1094" />
            </FindButton>

            <GoogleModal
                fillForm={fillForm}
                show={show}
                onHide={() => setShow(false)}
            />
        </div>
    );
}
