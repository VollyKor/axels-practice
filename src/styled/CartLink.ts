import styled from 'styled-components';

export const CartName = styled.span``;
export const CartQuantyty = styled.p``;

export const CartWrapper = styled.div`
    position: relative;
    color: var(--cart-link);
`;
export const CartText = styled.span``;

export const CartNotification = styled.div`
    position: absolute;
    top: 0;
    right: -2px;

    width: 12px;
    height: 12px;

    color: var(--white-color);
    background-color: var(--pink-color);
    border-radius: 50%50%;

    font-size: 9px;
`;
