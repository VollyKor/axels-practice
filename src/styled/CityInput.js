import styled from 'styled-components';

export const FindButton = styled.button`
    position: absolute;
    z-index: 1;

    right: 0;
    top: 4px;

    background-color: transparent;
    border: none;
`;
