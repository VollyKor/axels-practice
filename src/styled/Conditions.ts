import styled from 'styled-components';

export const ConditionText = styled.p`
    bottom: 0;
    position: initial;
    text-align: center;
    font-size: 0.6rem;

    @media ${({ theme }) => theme?.deviceSize?.conditionTextWidth} {
        position: absolute;
        margin-top: 0.8rem;
    } ;
`;

export const ConditionLink = styled.a`
    font-size: 0.65rem;
    color: var(--text-accent-color);
`;
