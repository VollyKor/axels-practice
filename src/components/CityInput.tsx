import { useState } from 'react';
import Form from 'react-bootstrap/Form';
import { BiTargetLock } from 'react-icons/bi';

import GoogleModal from 'components/GoodleModal';
import { FindButton } from 'styled/CityInput';

interface Props {
    inputRef: Ref;
    fillForm: TfillForm;
    value?: string;
    className?: string;
}

const CityInput = ({
    inputRef,
    fillForm,
    value = '',
    ...props
}: Props): JSX.Element => {
    const [show, setShow] = useState(false);
    return (
        <div className="position-relative">
            <Form.Group controlId="city" className="position-relative">
                <Form.Control
                    type="text"
                    placeholder="City"
                    ref={inputRef}
                    value={value}
                    {...props}
                />
            </Form.Group>

            <FindButton type="button" onClick={() => setShow(!show)}>
                <BiTargetLock size="25px" color="#5a1094" />
            </FindButton>

            <GoogleModal
                fillForm={fillForm}
                show={show}
                onHide={() => setShow(false)}
            />
        </div>
    );
};

export default CityInput;
