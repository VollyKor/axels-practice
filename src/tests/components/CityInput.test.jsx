import { shallow } from 'enzyme';

import CityInput from 'components/CityInput';
import GoodleModal from 'components/GoodleModal';
import { FindButton } from 'styled/CityInput';

it('renders snapshot', () => {
    const wrapper = shallow(<CityInput />);

    expect(wrapper).toMatchSnapshot();
});

it('changes visability of modal', () => {
    const wrapper = shallow(<CityInput />);

    expect(wrapper.find(GoodleModal).prop('show')).toEqual(false);

    wrapper.find(FindButton).simulate('click');
    expect(wrapper.find(GoodleModal).prop('show')).toEqual(true);
});

it('render props values correct', () => {
    const testValue = 'test value';
    const wrapper = shallow(<CityInput value={testValue} />);

    expect(wrapper.find('FormControl').prop('value')).toEqual(testValue);
    expect(wrapper.find('FormControl').prop('value')).not.toEqual('any value');

    wrapper.find(FindButton).simulate('click');
    expect(wrapper.find(GoodleModal).prop('show')).toEqual(true);
});

it('render default props values correct', () => {
    const wrapper = shallow(<CityInput />);

    expect(wrapper.find('FormControl').prop('value')).toEqual('');

    wrapper.find(FindButton).simulate('click');
    expect(wrapper.find(GoodleModal).prop('show')).toEqual(true);
});
