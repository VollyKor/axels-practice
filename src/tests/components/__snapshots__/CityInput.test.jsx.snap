// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot 1`] = `
<div
  className="position-relative"
>
  <FormGroup
    className="position-relative"
    controlId="city"
  >
    <FormControl
      placeholder="City"
      type="text"
      value=""
    />
  </FormGroup>
  <styled.button
    onClick={[Function]}
    type="button"
  >
    <BiTargetLock
      color="#5a1094"
      size="25px"
    />
  </styled.button>
  <GoogleModal
    onHide={[Function]}
    show={false}
  />
</div>
`;
